src/System.Xml.XPath.XmlDocument/src/System.Xml.XPath.XmlDocument.csproj

```diff
@@ -0,0 +1,20 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFrameworks>$(NetFrameworkMinimum);netstandard2.0;$(NetMinimum)</TargetFrameworks>
+    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+    <StrongNameKeyId>Open</StrongNameKeyId>
```

Comment:
Can you double check this public key token?  I'm seeing something different.  For that matter - make sure you're running APICompat by making this packable.

Reply:
> Can you double check this public key token?

I left it at "Open" as I could not find any StrongNameKeyId explicitly specified in the 1.1 codebase for this assembly.

> I'm seeing something different

Can you tell me what you tried? I might not know the technique you use to get the strong name key ID.

Also, did you double check you were doing System.Xml.XPath.**XmlDocument**? Just mentioning it because a couple of times I made the mistake of letting autocomplete choose "XDocument" instead of "XmlDocument".

> make sure you're running APICompat by making this packable.

Already did that before and I didn't get any warnings or errors when setting IsPackable=true.

Reply:
To check the strong name key I opened the old version of the package and opened the assembly.  I saw it was `System.Xml.XPath.XmlDocument, Version=4.0.2.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a`.  That `b03f5f7f11d50a3a` is `Microsoft` https://github.com/dotnet/arcade/blob/85ae8b264f3312aac5ca08dcab039556e99fcb00/src/Microsoft.DotNet.Arcade.Sdk/tools/StrongName.targets#L31-L37

Reply:
I changed it to Microsoft, then ran dotnet pack again with IsPackable set to true, and there are still no ApiCompat problems. I ran it with /bl and I do see a bunch of ApiCompat related messages, so it's running. I can't pinpoint the exact place where the comparisons are done, though. Where should I look exactly? (see attached)
[binlog.zip](https://github.com/dotnet/maintenance-packages/files/14454866/binlog.zip)

Reply:
I can confirm there are no ApiCompat diffs to resolve. I manually altered a random public API signature, ran dotnet pack again, and I can see that it is detecting the difference:

```
dotnet pack .\src\System.Xml.XPath.XmlDocument\src\System.Xml.XPath.XmlDocument.csproj
MSBuild version 17.9.4+90725d08d for .NET
  Determining projects to restore...
  All projects are up-to-date for restore.
C:\Users\calope\source\repos\maintenance-packages\src\System.Xml.XPath.XmlDocument\src\System\Xml\DocumentXPathNavigator.cs(1513,44): error CS7036: There is no argument given that corresponds to the r
equired parameter 'x' of 'DocumentXmlWriter.DocumentXmlWriter(DocumentXmlWriterType, XmlNode, XmlDocument, bool)' [C:\Users\calope\source\repos\maintenance-packages\src\System.Xml.XPath.XmlDocument\sr
c\System.Xml.XPath.XmlDocument.csproj::TargetFramework=net462]
C:\Users\calope\source\repos\maintenance-packages\src\System.Xml.XPath.XmlDocument\src\System\Xml\DocumentXPathNavigator.cs(1530,44): error CS7036: There is no argument given that corresponds to the r
equired parameter 'x' of 'DocumentXmlWriter.DocumentXmlWriter(DocumentXmlWriterType, XmlNode, XmlDocument, bool)' [C:\Users\calope\source\repos\maintenance-packages\src\System.Xml.XPath.XmlDocument\sr
c\System.Xml.XPath.XmlDocument.csproj::TargetFramework=net462]
C:\Users\calope\source\repos\maintenance-packages\src\System.Xml.XPath.XmlDocument\src\System\Xml\DocumentXPathNavigator.cs(1555,44): error CS7036: There is no argument given that corresponds to the r
equired parameter 'x' of 'DocumentXmlWriter.DocumentXmlWriter(DocumentXmlWriterType, XmlNode, XmlDocument, bool)' [C:\Users\calope\source\repos\maintenance-packages\src\System.Xml.XPath.XmlDocument\sr
c\System.Xml.XPath.XmlDocument.csproj::TargetFramework=net462]
C:\Users\calope\source\repos\maintenance-packages\src\System.Xml.XPath.XmlDocument\src\System\Xml\DocumentXPathNavigator.cs(1579,44): error CS7036: There is no argument given that corresponds to the r
equired parameter 'x' of 'DocumentXmlWriter.DocumentXmlWriter(DocumentXmlWriterType, XmlNode, XmlDocument, bool)' [C:\Users\calope\source\repos\maintenance-packages\src\System.Xml.XPath.XmlDocument\sr
c\System.Xml.XPath.XmlDocument.csproj::TargetFramework=net462]
C:\Users\calope\source\repos\maintenance-packages\src\System.Xml.XPath.XmlDocument\src\System\Xml\DocumentXPathNavigator.cs(1591,44): error CS7036: There is no argument given that corresponds to the r
equired parameter 'x' of 'DocumentXmlWriter.DocumentXmlWriter(DocumentXmlWriterType, XmlNode, XmlDocument, bool)' [C:\Users\calope\source\repos\maintenance-packages\src\System.Xml.XPath.XmlDocument\sr
c\System.Xml.XPath.XmlDocument.csproj::TargetFramework=net462]
C:\Users\calope\source\repos\maintenance-packages\src\System.Xml.XPath.XmlDocument\src\System\Xml\DocumentXPathNavigator.cs(1647,44): error CS7036: There is no argument given that corresponds to the r
equired parameter 'x' of 'DocumentXmlWriter.DocumentXmlWriter(DocumentXmlWriterType, XmlNode, XmlDocument, bool)' [C:\Users\calope\source\repos\maintenance-packages\src\System.Xml.XPath.XmlDocument\sr
c\System.Xml.XPath.XmlDocument.csproj::TargetFramework=net462]
C:\Users\calope\source\repos\maintenance-packages\src\System.Xml.XPath.XmlDocument\src\System\Xml\DocumentXPathNavigator.cs(1513,44): error CS7036: There is no argument given that corresponds to the r
equired parameter 'x' of 'DocumentXmlWriter.DocumentXmlWriter(DocumentXmlWriterType, XmlNode, XmlDocument, bool)' [C:\Users\calope\source\repos\maintenance-packages\src\System.Xml.XPath.XmlDocument\sr
c\System.Xml.XPath.XmlDocument.csproj::TargetFramework=netstandard2.0]
C:\Users\calope\source\repos\maintenance-packages\src\System.Xml.XPath.XmlDocument\src\System\Xml\DocumentXPathNavigator.cs(1530,44): error CS7036: There is no argument given that corresponds to the r
equired parameter 'x' of 'DocumentXmlWriter.DocumentXmlWriter(DocumentXmlWriterType, XmlNode, XmlDocument, bool)' [C:\Users\calope\source\repos\maintenance-packages\src\System.Xml.XPath.XmlDocument\sr
c\System.Xml.XPath.XmlDocument.csproj::TargetFramework=netstandard2.0]
C:\Users\calope\source\repos\maintenance-packages\src\System.Xml.XPath.XmlDocument\src\System\Xml\DocumentXPathNavigator.cs(1555,44): error CS7036: There is no argument given that corresponds to the r
equired parameter 'x' of 'DocumentXmlWriter.DocumentXmlWriter(DocumentXmlWriterType, XmlNode, XmlDocument, bool)' [C:\Users\calope\source\repos\maintenance-packages\src\System.Xml.XPath.XmlDocument\sr
c\System.Xml.XPath.XmlDocument.csproj::TargetFramework=netstandard2.0]
C:\Users\calope\source\repos\maintenance-packages\src\System.Xml.XPath.XmlDocument\src\System\Xml\DocumentXPathNavigator.cs(1579,44): error CS7036: There is no argument given that corresponds to the r
equired parameter 'x' of 'DocumentXmlWriter.DocumentXmlWriter(DocumentXmlWriterType, XmlNode, XmlDocument, bool)' [C:\Users\calope\source\repos\maintenance-packages\src\System.Xml.XPath.XmlDocument\sr
c\System.Xml.XPath.XmlDocument.csproj::TargetFramework=netstandard2.0]
C:\Users\calope\source\repos\maintenance-packages\src\System.Xml.XPath.XmlDocument\src\System\Xml\DocumentXPathNavigator.cs(1591,44): error CS7036: There is no argument given that corresponds to the r
equired parameter 'x' of 'DocumentXmlWriter.DocumentXmlWriter(DocumentXmlWriterType, XmlNode, XmlDocument, bool)' [C:\Users\calope\source\repos\maintenance-packages\src\System.Xml.XPath.XmlDocument\sr
c\System.Xml.XPath.XmlDocument.csproj::TargetFramework=netstandard2.0]
C:\Users\calope\source\repos\maintenance-packages\src\System.Xml.XPath.XmlDocument\src\System\Xml\DocumentXPathNavigator.cs(1647,44): error CS7036: There is no argument given that corresponds to the r
equired parameter 'x' of 'DocumentXmlWriter.DocumentXmlWriter(DocumentXmlWriterType, XmlNode, XmlDocument, bool)' [C:\Users\calope\source\repos\maintenance-packages\src\System.Xml.XPath.XmlDocument\sr
c\System.Xml.XPath.XmlDocument.csproj::TargetFramework=netstandard2.0]
C:\Users\calope\source\repos\maintenance-packages\src\System.Xml.XPath.XmlDocument\src\System\Xml\DocumentXPathNavigator.cs(1513,44): error CS7036: There is no argument given that corresponds to the r
equired parameter 'x' of 'DocumentXmlWriter.DocumentXmlWriter(DocumentXmlWriterType, XmlNode, XmlDocument, bool)' [C:\Users\calope\source\repos\maintenance-packages\src\System.Xml.XPath.XmlDocument\sr
c\System.Xml.XPath.XmlDocument.csproj::TargetFramework=net6.0]
C:\Users\calope\source\repos\maintenance-packages\src\System.Xml.XPath.XmlDocument\src\System\Xml\DocumentXPathNavigator.cs(1530,44): error CS7036: There is no argument given that corresponds to the r
equired parameter 'x' of 'DocumentXmlWriter.DocumentXmlWriter(DocumentXmlWriterType, XmlNode, XmlDocument, bool)' [C:\Users\calope\source\repos\maintenance-packages\src\System.Xml.XPath.XmlDocument\sr
c\System.Xml.XPath.XmlDocument.csproj::TargetFramework=net6.0]
C:\Users\calope\source\repos\maintenance-packages\src\System.Xml.XPath.XmlDocument\src\System\Xml\DocumentXPathNavigator.cs(1555,44): error CS7036: There is no argument given that corresponds to the r
equired parameter 'x' of 'DocumentXmlWriter.DocumentXmlWriter(DocumentXmlWriterType, XmlNode, XmlDocument, bool)' [C:\Users\calope\source\repos\maintenance-packages\src\System.Xml.XPath.XmlDocument\sr
c\System.Xml.XPath.XmlDocument.csproj::TargetFramework=net6.0]
C:\Users\calope\source\repos\maintenance-packages\src\System.Xml.XPath.XmlDocument\src\System\Xml\DocumentXPathNavigator.cs(1579,44): error CS7036: There is no argument given that corresponds to the r
equired parameter 'x' of 'DocumentXmlWriter.DocumentXmlWriter(DocumentXmlWriterType, XmlNode, XmlDocument, bool)' [C:\Users\calope\source\repos\maintenance-packages\src\System.Xml.XPath.XmlDocument\sr
c\System.Xml.XPath.XmlDocument.csproj::TargetFramework=net6.0]
C:\Users\calope\source\repos\maintenance-packages\src\System.Xml.XPath.XmlDocument\src\System\Xml\DocumentXPathNavigator.cs(1591,44): error CS7036: There is no argument given that corresponds to the r
equired parameter 'x' of 'DocumentXmlWriter.DocumentXmlWriter(DocumentXmlWriterType, XmlNode, XmlDocument, bool)' [C:\Users\calope\source\repos\maintenance-packages\src\System.Xml.XPath.XmlDocument\sr
c\System.Xml.XPath.XmlDocument.csproj::TargetFramework=net6.0]
C:\Users\calope\source\repos\maintenance-packages\src\System.Xml.XPath.XmlDocument\src\System\Xml\DocumentXPathNavigator.cs(1647,44): error CS7036: There is no argument given that corresponds to the r
equired parameter 'x' of 'DocumentXmlWriter.DocumentXmlWriter(DocumentXmlWriterType, XmlNode, XmlDocument, bool)' [C:\Users\calope\source\repos\maintenance-packages\src\System.Xml.XPath.XmlDocument\sr
c\System.Xml.XPath.XmlDocument.csproj::TargetFramework=net6.0]
```

Reply:
> Where should I look exactly? (see attached)

In that attached log it shows APICompat is not running - it's skipped.
<img width="813" alt="image" src="https://github.com/dotnet/maintenance-packages/assets/8918108/a4821ca7-469c-47b1-a6b2-7fdaa32eab41">

> I manually altered a random public API signature, ran dotnet pack again, and I can see that it is detecting the difference:

It's good that you can force it to run, the problem we were looking at here was "Why doesn't API Compat catch the assembly name mismatch".  If you can prove that it's not catching the mismatch, then that's a bug that we need to file.  If instead it's just that we don't have a good handle over when APICompat runs - then that's something to fix in the dev process so that you can better rely on this helpful tool.

Reply:
Huh... Okay I tried it again. I set the key to "Open", IsPackable to true, and ran dotnet pack on the src XmlDocument.csproj. It still passes. Maybe it's a bug in ApiCompat and it should've failed when detecting "Open":

![image](https://github.com/dotnet/maintenance-packages/assets/1175054/06a93376-f433-404a-a209-987f0655ce9f)